WcaOnRails/app/models/competition.rb

```diff
@@ -305,6 +305,18 @@ def events
     eventSpecs.split.map { |e| Event.find_by_id(e.split("=")[0]) }.sort_by &:rank
   end
 
+  def has_event?(event_id)
```

Comment:
This method is called `has_event?`, so I'd expect it to take an `Event`, not an `event_id`. How about either changing the name of the method, or changing it to take an actual Event? Whichever is cleaner.